File: Java/JavaNapiSample/src/com/nymi/api/sample/ProvisionStorage.java
package com.nymi.api.sample;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class ProvisionStorage {

	String filename;
	
	ProvisionStorage (String fn) {
		filename = fn;
	}
	
	String read() {
		try {
			BufferedReader br = new BufferedReader(new FileReader(filename));
			String provisions = br.readLine();
			br.close();
			if (provisions.isEmpty()) return "{}";
			return provisions;
		}
		catch (Exception e) {
			return "{}";
		}
	}
	
	boolean write(String provisions) {
		try {
			BufferedWriter bw = new BufferedWriter(new FileWriter(filename));
			bw.write(provisions);
			bw.close();
			return true;
		}
		catch (IOException e) {
			System.out.println("ERROR: Unable to writing provision file: " + e.getLocalizedMessage());
			return false;
		}
	}
}

File: Cplusplus/practNapi/src/ProvisionStorage.cpp
#include "ProvisionStorage.h"

#include <fstream>

using namespace std;

ProvisionStorage::ProvisionStorage(std::string filename): m_filename(filename) {
	// Nothing else
}

ProvisionStorage::~ProvisionStorage() {
}

std::string ProvisionStorage::read() {
	std::ifstream fs(m_filename);
	if (!fs.is_open()) {
		return "{}";
	}

	std::string jsonOut = "";
	std::string linein;

	while (getline(fs, linein))
		jsonOut.append(linein);

	fs.close();
	return jsonOut;
}

bool ProvisionStorage::write(std::string provisions) {
	std::ofstream fs(m_filename);
	if (!fs.is_open()) {
		std::cout << "ERROR: Unable to open provision file " << m_filename << " to write provisions" << endl;
		return false;
	}

	fs << provisions << endl;

	fs.close();

	return true;
}
